packages/react/src/components/Forms/Select/mono-simple.tsx

```diff
@@ -23,7 +23,7 @@ export const SelectMonoSimple = (props: SubProps) => {
       (option) => optionToValue(option) === props.value,
     );
     downshiftReturn.selectItem(optionToSelect ?? null);
-  }, [props.value, props.options, props.downshiftProps]);
+  }, [props.value, props.options, props.downshiftProps.initialSelectedItem]);
```

Comment:
If you could do another commit for just for this as it does not play any role fixing the infinite loop. ( Only `setValue` does )